beacon-chain/rpc/prysm/v1alpha1/validator/proposer_capella.go

```diff
@@ -0,0 +1,25 @@
+package validator
+
+import (
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	"github.com/prysmaticlabs/prysm/v3/consensus-types/interfaces"
+	"github.com/prysmaticlabs/prysm/v3/time/slots"
+)
+
+// Sets the bls to exec data for a block.
+func (vs *Server) setBlsToExecData(blk interfaces.BeaconBlock, headState state.BeaconState) error {
+	if slots.ToEpoch(blk.Slot()) < params.BeaconConfig().CapellaForkEpoch {
+		return nil
+	}
+
+	changes, err := vs.BLSChangesPool.BLSToExecChangesForInclusion(headState)
+	if err != nil {
+		log.WithError(err).Error("Could not get bls to execution changes")
```

Comment:
If we log this here and then return nil below, wil the slice be nil or empty? we want to have an empty slice here.

Reply:
how are we setting an empty slice in this path?

Reply:
It's L15 no? 

```go
if err := blk.Body().SetBLSToExecutionChanges([]*ethpb.SignedBLSToExecutionChange{}); err != nil {
		log.WithError(err).Error("Could not set bls to execution data in block")
		return
	}
```